aips/code-verification-api.md

```diff
@@ -0,0 +1,126 @@
+This API proposal seeks to introduce a standard protocol for the verification of Aptos Move smart contract code. This API is designed to ensure the safety and trustworthiness of smart contract code, assisting developers in effectively verifying their code. The standard suggests the common rules that should be followed by all parties building and operating smart contract code verification API servers within the Aptos network.
+
+## Motivation
+
+Blockchain is fundamentally systems for safely storing and transferring valuable assets. Many assets are managed by smart contracts, and the code of these smart contracts dictates how these assets move and are stored. Therefore, it is extremely important to verify the code of these smart contracts.
+
+Without such a verification process, serious consequences could arise from code written incorrectly due to developer error or malicious intent. These consequences can range from the loss of specific assets to threatening the stability of the entire network.
+
+Of course, code verification can be performed locally through the SDK. However, to increase accessibility and usability, it's most desirable to provide it through an API in the explorer.
+
+In this proposal, I aim to address these issues by introducing a smart contract code verification API. This API will assist developers in confirming the code with which they have deployed their smart contracts. This will enhance the safety and reliability of the code and enable potential problems to be discovered and corrected in advance.
+
+## Impact
+
+This change will impact smart contract developers and users of the Aptos network. Developers will be able to use this API to verify their code, and users can gain confidence in the safety of the verified smart contracts.
+
+When users know that a smart contract has been verified through this API, they gain trust that they can use that smart contract more safely. This enables users to engage in transactions and interactions within the network with greater confidence.
+
+Therefore, this change will contribute to enhancing the overall stability and trustworthiness of the Aptos network.
+
+## Rationale
+
+This proposal is the most effective way to improve the stability and trustworthiness of the Aptos network. This API will help developers make smart contracts safer and play a crucial role in enhancing the network's transparency and increasing user trust. Since Aptos is designed to store the code and the compiled bytecode on-chain when deploying Move code, verification is more convenient compared to other chains.
+
+## Specification
+
+### **Request**
+
+For verification of the smart contract code, developers must send a **`POST`** request to the API endpoint. It must include the following fields:
+
+- **`address`**: The address where the smart contract to be verified is deployed.
+- **`moduleName`**: The name of the module to be verified.
+- **`compilerVersion`**: The version of the Move compiler used for compilation.
```

Comment:
This is not needed, it should be part of the package metadata which is on chain.